docs/source/contributor-guide/parquet_scans.md

```diff
@@ -131,7 +131,505 @@ The S3 support of `native_datafusion` has the following limitations:
+                                        │                           │
+                                        │ Configuration translation:│
+                                        │ Hadoop S3A configs →      │
+                                        │ object_store configs      │
+                                        │                           │
+                                        │ - fs.s3a.access.key       │
+                                        │ - fs.s3a.secret.key       │
+                                        │ - fs.s3a.endpoint         │
+                                        │ - fs.s3a.endpoint.region  │
+                                        └───────────────────────────┘
+
+Key Characteristics:
+- ✅ Fully native execution (no JVM→Native data transfer for scan)
+- ✅ Complex type support (structs, arrays, maps)
+- ✅ Leverages DataFusion community improvements
+- ✅ Better performance for some workloads
+- ❌ More restrictions than native_comet (no bucketed scans)
+- ❌ Cannot read UINT_8/UINT_16 types by default
+- ❌ No support for ignoreMissingFiles/ignoreCorruptFiles
+- ⚠️  Requires COMET_EXEC_ENABLED=true
+- ⚠️  Uses object_store (different from Hadoop FileSystem)
+```
+
+### `native_iceberg_compat` Scan Architecture
+
+The `native_iceberg_compat` scan is designed for Iceberg integration, similar to `native_datafusion` but with an Iceberg-specific API layer.
+
+```
+┌─────────────────────────────────────────────────────────────────────────────┐
+│                   NATIVE_ICEBERG_COMPAT SCAN FLOW                           │
+└─────────────────────────────────────────────────────────────────────────────┘
+
+JVM (Scala/Java)                                    Native (Rust)
+════════════════════════════════════════════════════════════════════════════════
+
+┌───────────────────────────────┐
+│ CometScanExec                 │  Physical operator for DataSource V1
+│ (CometScanExec.scala:61)      │
+│                               │  Auto-selected for Iceberg tables
+│ scanImpl =                    │  by CometScanRule.selectScan()
+│   "native_iceberg_compat"     │  (CometScanRule.scala:357)
+│                               │
+│ Key difference:               │
+│ - Prefetch is DISABLED        │  Iceberg manages its own data access
+│   (line 434: !usingDataFusion │  patterns
+│    Reader)                    │
+└───────────┬───────────────────┘
+            │
+            │ Uses Iceberg-specific reader
+            ↓
+┌───────────────────────────────┐
+│ IcebergCometBatchReader       │  Public API for Iceberg integration
+│ (IcebergCometBatchReader.java │
+│  :29-43)                      │
+│                               │
+│ Extends: BatchReader          │  Inherits batch reading functionality
+│                               │
+│ Key method:                   │
+│ init(AbstractColumnReader[])  │  Iceberg provides column readers
+│                               │
+│ Purpose:                      │
```

Comment:
This is not correct. (This is how the current integration of native_comet and Iceberg works). `native_iceberg_compat` uses the same `init_datasource_exec` call to create a DataFusion `DataSourceExec` and wraps the natve batch and native columns in corresponding classes on the JVM side (NativeBatchReader and NativeColumnReader)